storage/client/client.go

```diff
@@ -610,22 +712,38 @@ func (c *StorageClient) Delegations(ctx context.Context, r *DelegationsRequest,
 	}
 	for rows.Next() {
 		var d Delegation
-		var escrowBalanceActive uint64
-		var escrowTotalSharesActive uint64
+		var sharesNum pgtype.Numeric
+		var escrowBalanceActiveNum pgtype.Numeric
+		var escrowTotalSharesActiveNum pgtype.Numeric
 		if err := rows.Scan(
 			&d.ValidatorAddress,
-			&d.Shares,
-			&escrowBalanceActive,
-			&escrowTotalSharesActive,
+			&sharesNum,
+			&escrowBalanceActiveNum,
+			&escrowTotalSharesActiveNum,
 		); err != nil {
 			c.logger.Info("row scan failed",
 				"request_id", ctx.Value(RequestIDContextKey),
 				"err", err.Error(),
 			)
 			return nil, common.ErrStorageError
 		}
-
-		d.Amount = uint64(float64(d.Shares) * float64(escrowBalanceActive) / float64(escrowTotalSharesActive))
+		shares, err := c.NumericToBigInt(ctx, &sharesNum)
+		if err != nil {
+			return nil, common.ErrStorageError
+		}
+		escrowBalanceActive, err := c.NumericToBigInt(ctx, &escrowBalanceActiveNum)
+		if err != nil {
+			return nil, common.ErrStorageError
+		}
+		escrowTotalSharesActive, err := c.NumericToBigInt(ctx, &escrowBalanceActiveNum)
+		if err != nil {
+			return nil, common.ErrStorageError
+		}
+		amount := big.NewInt(0)
+		amount.Mul(shares, escrowBalanceActive)
```

Comment:
Slightly nicer, here and elsewhere:
```suggestion
		amount := new(big.Int).Mul(shares, escrowBalanceActive)
```
What a janky API though ...